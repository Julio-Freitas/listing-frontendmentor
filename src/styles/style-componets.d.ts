import { Theme } from './theme';

type Theme = typeof Theme;
declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
