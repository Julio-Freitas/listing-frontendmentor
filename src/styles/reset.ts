import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    // this is the shared style
    html, body {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: ${({ theme }) => theme.fonts.primary.leagueSpartan};
    background-color:${({ theme }) =>
      theme.colors.secondary.lightGrayishCyanBackground};
    margin: 0 auto;

    }

    #__next {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      max-width: 1140px;
      margin: 0 auto;
    }

    #__next main {
        display: flex;
        flex-direction: column;
        height: 100%;
        width: 100%;
    }
    * {
        box-sizing: border-box;
        margin: 0;
        padding: 0;
    }
    a{
        text-decoration: none;
    }

    input {
        all: unset;
    }

    button {
        cursor: pointer
    }
    img{
      display: inline-block;
      max-width:100%;
      margin:0;
    }
`;
