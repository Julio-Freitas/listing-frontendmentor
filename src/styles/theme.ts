export const Theme = {
  fonts: {
    primary: {
      leagueSpartan: 'League Spartan, sans-serif'
    },
    secondary: {
      lexendDeca: 'Lexend Deca, sans-serif'
    }
  },
  colors: {
    primary: {
      darkCyan: 'hsl(180, 29%, 50%)'
    },
    secondary: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilter: 'hsl(180, 31%, 95%)',
      darkGrayishCyanFilter: 'hsl(180, 8%, 52%)',
      veryDarkGrayinshCyan: 'hsl(180, 14%, 20%)'
    }
  }
};
