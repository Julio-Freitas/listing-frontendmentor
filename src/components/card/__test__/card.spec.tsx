import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Theme } from '../../../styles/theme';
import { Card } from '..';

const mockedProps = {
  id: 1,
  company: 'Photosnap',
  logo: '/images/photosnap.svg',
  new: true,
  featured: true,
  position: 'Senior Frontend Developer',
  role: 'Frontend',
  level: 'Senior',
  postedAt: '1d ago',
  contract: 'Full Time',
  location: 'USA Only',
  languages: ['HTML', 'CSS', 'JavaScript'],
  tools: [],
  onClickLanguage: () => false
};

const renderWithProviderTheme = (component: React.ReactNode) => {
  return {
    ...render(<ThemeProvider theme={Theme}>{component}</ThemeProvider>)
  };
};

describe('<Card />', () => {
  it('should be render correctly', () => {
    const { container } = renderWithProviderTheme(
      <Card {...mockedProps} isFeatured isNew />
    );
    expect(container.firstChild).toMatchSnapshot();

    const skills = screen.getByTestId('skills');
    const newEl = screen.getByTestId('new');

    expect(screen.getByText('Photosnap')).toBeInTheDocument();
    expect(screen.getByText('Senior Frontend Developer')).toBeInTheDocument();
    expect(screen.getByText('1d ago')).toBeInTheDocument();
    expect(screen.getByText('Full Time')).toBeInTheDocument();
    expect(screen.getByText('USA Only')).toBeInTheDocument();

    expect(newEl).toBeInTheDocument();
    expect(skills.childNodes.length).toBe(mockedProps.languages.length);
  });

  it('should be render correctly info new is false', () => {
    renderWithProviderTheme(<Card {...mockedProps} isNew={false} isFeatured />);

    const newEl = screen.queryByTestId('new');
    expect(newEl).toBeNull();
  });

  it('should be render correctly info featured is false', () => {
    renderWithProviderTheme(<Card {...mockedProps} isFeatured={false} isNew />);

    const featuredEl = screen.queryByTestId('featured');
    expect(featuredEl).toBeNull();
  });
});
