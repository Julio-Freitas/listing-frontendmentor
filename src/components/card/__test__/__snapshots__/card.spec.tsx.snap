// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should be render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 5px;
  padding: 25px;
  border: 5px transparent solid;
  border-left-color: hsl(180,29%,50%);
  height: 115px;
  background: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
  padding-left: 25px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 h2 {
  font-size: 1.6rem;
}

.c2 {
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  gap: 15px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 label {
  font-size: 1.4rem;
  color: hsl(180,29%,50%);
  font-weight: 600;
}

.c3 {
  font-size: 1.2rem;
  background: hsl(180,29%,50%);
  padding: 0.4rem 0.5rem;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3::before {
  content: attr(data-text);
  text-transform: uppercase;
  color: hsl(180,31%,95%);
  position: relative;
  top: 1px;
}

.c4 {
  font-size: 1.2rem;
  background: hsl(180,14%,20%);
  padding: 0.4rem 0.5rem;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4::before {
  content: attr(data-text);
  text-transform: uppercase;
  color: hsl(180,31%,95%);
  position: relative;
  top: 1px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 190px;
}

.c6 {
  font-size: 1.2rem;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6:first-child {
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  gap: 25%;
}

.c6:after {
  content: ' ';
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: .4rem;
  height: .4rem;
  border-radius: 50%;
}

.c6:after {
  background: hsl(180,8%,52%);
}

.c7 {
  font-size: 1.2rem;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:first-child {
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  gap: 25%;
}

.c7:after {
  background: hsl(180,8%,52%);
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  gap: 15px;
}

.c9 {
  font-size: 1.5rem;
  background: hsl(180,52%,96%);
  color: hsl(180,29%,50%);
  font-weight: 600;
  padding: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  border-radius: 0.3rem;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c9:hover {
  background: hsl(180,29%,50%);
  color: hsl(180,52%,96%);
}

<li
  class="c0"
  role="listitem"
>
  <img
    alt="image card"
    data-nimg="1"
    decoding="async"
    height="60"
    loading="lazy"
    src="/images/photosnap.svg"
    srcset="/images/photosnap.svg 1x, /images/photosnap.svg 2x"
    style="color: transparent;"
    width="60"
  />
  <div
    class="c1"
  >
    <header
      class="c2"
    >
      <label>
        Photosnap
      </label>
      <span
        class="c3"
        data-testid="new"
        data-text="new!"
      />
      <span
        class="c4"
        data-testid="isFeatured"
        data-text="isFeatured!"
      />
    </header>
    <h2>
      Senior Frontend Developer
    </h2>
    <footer
      class="c5"
    >
      <span
        class="c6"
      >
        1d ago
      </span>
      <span
        class="c6"
      >
        Full Time
         
      </span>
      <span
        class="c7"
      >
        USA Only
      </span>
    </footer>
  </div>
  <div
    class="c8"
    data-testid="skills"
  >
    <div
      class="c9"
    >
      HTML
    </div>
    <div
      class="c9"
    >
      CSS
    </div>
    <div
      class="c9"
    >
      JavaScript
    </div>
  </div>
</li>
`;
